src/test/perl/path.t

```diff
@@ -290,39 +290,68 @@ rmtree if -d $basetest;
 ok(! $mc->directory_exists($basetest), "directory_exists false on missing directory");
 ok(! $mc->file_exists($basetest), "file_exists false on missing directory");
 ok(! $mc->any_exists($basetest), "any_exists false on missing directory");
+ok(! $mc->is_symlink($basetest), "is_symlink false on missing directory");
 
 ok(! $mc->directory_exists($basetestfile), "directory_exists false on missing file");
 ok(! $mc->file_exists($basetestfile), "file_exists false on missing file");
 ok(! $mc->any_exists($basetestfile), "any_exists false on missing file");
+ok(! $mc->is_symlink($basetestfile), "is_symlink false on missing file");
 
 makefile($basetestfile);
 
 ok($mc->directory_exists($basetest), "directory_exists true on created directory");
 ok($mc->any_exists($basetest), "any_exists true on created directory");
 ok(! $mc->file_exists($basetest), "file_exists false on created directory");
+ok(! $mc->is_symlink($basetest), "is_symlink false on created directory");
 
 ok(! $mc->directory_exists($basetestfile), "directory_exists false on created file");
 ok($mc->any_exists($basetestfile), "any_exists true on created file");
 ok($mc->file_exists($basetestfile), "file_exists true on created file");
-
-# Test (broken) symlink and _exsists methods
-
-ok(symlink("really_really_missing", $brokenlink), "broken symlink created");
-makefile("$basetest/tgtdir/tgtfile");
-ok(symlink("tgtdir", $dirlink), "directory symlink created");
-ok(symlink("tgtdir/tgtfile", $filelink), "file symlink created");
+ok(! $mc->is_symlink($basetestfile), "is_symlink false on created file");
+
+# Test symlink creation
+
+my $target_directory = "tgtdir";
+my $target_file1 = "tgtfile1";
+my $target_file2 = "$basetest/$target_directory/tgtfile2";
+makefile("$basetest/$target_file1");
+makefile($target_file2);
+my %opts;
+is($mc->symlink($dirlink, $target_directory), CHANGED, "directory symlink created");
+is($mc->symlink($filelink, $target_file2), CHANGED, "file symlink created");
+is($mc->symlink($filelink, $target_file2), SUCCESS, "file symlink already exists: nothing done");
+is($mc->symlink($filelink, $target_file1), CHANGED, "file symlink already exists: nothing done");
+my $link_status = $mc->symlink($target_file2, $target_file1);
+ok(! $link_status, "existing file not replaced by a symlink");
+is($mc->{fail}, "Failed to create symlink $target_file2 (target=$target_file1)");
+$opts{force} = 1;
+is($mc->symlink($target_file2, $target_file1, %opts), CHANGED, "existing file replaced by a symlink (force option set)");
+ok(! $mc->symlink("$basetest/$target_directory", $target_file1, %opts), "directory not replaced by a symlink (force option set)");
```

Comment:
ok, i reread the pod, and this is implied, but it would be better to make it explicit in the pod (or use the not replacing of a directory as an exmaple in the pod).